### PR TITLE
fix emoji picker search field not focused nor usable

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/10990

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export const useSpaceActionsSetIcon = () => {
       title: $gettext('Set icon for %{space}', { space: resources[0].name }),
       hideConfirmButton: true,
       customComponent: EmojiPickerModal,
+      focusTrapActive: false,
```
Comment (kulmann) [inline packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts:34]: Ooooh wait. The focus trap is required for all modals, otherwise it's an a11y violation. Why is this line required?

Comment (AlexAndBear) [reply packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts:34]: This is JUST for the emoji picker modal, the emoji picker is not compatible with the shadowRoot implementation of the emoji picker and so the emoji picker handles the focus by itself. What does the focus trap in addition ?

Comment (AlexAndBear) [reply packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts:34]: Ok understood what the trap should do, somehow the focus trap breaks the emoji picker ...

Comment (AlexAndBear) [reply packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts:34]: Got it, I don't have a solution to this on hand rn

Comment (AlexAndBear) [reply packages/web-pkg/src/composables/actions/spaces/useSpaceActionsSetIcon.ts]: Fixed

File: packages/design-system/src/components/OcModal/OcModal.vue [hunk 2]
```diff
@@ -302,6 +302,13 @@ export default defineComponent({
     const showSpinner = ref(false)
     const buttonConfirmAppearance = ref('filled')
 
+    const tabbableOptions = computed((): FocusTrapTabbableOptions => {
+      // Enable shadowDom support for e.G emoji-picker
```
Comment (dschmidt) [inline packages/design-system/src/components/OcModal/OcModal.vue:306]:
```suggestion
      // Enable shadow DOM support for e.g. emoji-picker
```

**Review by dschmidt — APPROVED** [review]

**Review by kulmann — APPROVED** [review]
Thank you 😍

File: packages/design-system/src/components/OcModal/OcModal.vue [hunk 3]
```diff
@@ -340,8 +351,8 @@ export default defineComponent({
   },
   computed: {
     initialFocusRef(): FocusTargetOrFalse {
-      if (this.focusTrapInitial) {
-        return this.focusTrapInitial
+      if (this.focusTrapInitial || this.focusTrapInitial === false) {
+        return this.focusTrapInitial as FocusTargetOrFalse
```
Comment (dschmidt) [inline packages/design-system/src/components/OcModal/OcModal.vue:355]: Why not use `FocusTargetOrFalse` as type for `focusTrapInitial` directly?
